server/sonar-server/src/main/java/org/sonar/server/computation/taskprocessor/CeProcessingSchedulerImpl.java

```diff
@@ -48,39 +54,74 @@ public CeProcessingSchedulerImpl(CeProcessingSchedulerExecutorService processing
   }
 
   @Override
+  public void start() {
+    // nothing to do at component startup, startScheduling will be called by CeQueueInitializer
+  }
+
+  @Override
   public void startScheduling() {
     ListenableScheduledFuture<Boolean> future = executorService.schedule(workerRunnable, delayBetweenTasks, timeUnit);
 
-    FutureCallback<Boolean> chainingCallback = new ChainingCallback();
     addCallback(future, chainingCallback, executorService);
   }
 
+  @Override
+  public void stop() {
+    this.chainingCallback.stop();
+  }
+
   private class ChainingCallback implements FutureCallback<Boolean> {
+    private final AtomicBoolean keepRunning = new AtomicBoolean(true);
+    @CheckForNull
+    private ListenableFuture<Boolean> workerFuture;
+
     @Override
     public void onSuccess(@Nullable Boolean result) {
```

Comment:
a non nullable `Boolean` would be a `boolean` but it can not be because of generics, so I stick with the signatore defined by `FutureCallBack` and keep the `@Nullable`